src/rect.rs

```diff
@@ -52,6 +53,43 @@ impl<T: PartialOrd + Clone, Unit> Rect<T, Unit> {
     }
 }
 
+impl<T, U, V, Unit> Rect<T, Unit>
+where
+    T: Sub<T, Output = U> + Add<V, Output = T> + Clone,
+    U: Mul<f32, Output = V>,
```

Comment:
Can we do this without mentioning `f32`? I think that pretty much restricts these functions to working on `Rect<f32, U>`'s.

Reply:
Yeah that's actually a big question I had while doing this, which I forgot about until now. How to do the `0.5` while remaining generic.

Should I write `F` instead of `f32` and then do something like `0.5 as F` in the functions below? Probably not exactly that, but something in the spirit of that?

Reply:
I think there's no way 0.5 will work for ints.

This is how `euclid` does it: https://doc.servo.org/src/euclid/rect.rs.html#351-359
Basically divide by two instead.

This crate has the `One` trait in it: https://github.com/rust-num/num-traits

Reply:
Got it. That seems like a reasonable solution. I'll play with that